### PR TITLE
Add arithmetic and data management instructions

### PR DESCRIPTION
**Pull request by MMagueta**

### CODE REVIEW

File: Chime/Instructions.m [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+void instruction_op_and(Stack *_dataStack){
+    NSInteger value1 = [[_dataStack pop] integerValue];
+    NSInteger value2 = [[_dataStack pop] integerValue];
+    [_dataStack push:@(value1 & value2)];
+    return;
+}
+
+void instruction_op_or(Stack *_dataStack){ NSInteger value1 = [[_dataStack pop] integerValue];
```
Comment (EduardoLR10) [inline Chime/Instructions.m]:
```suggestion
void instruction_op_or(Stack *_dataStack){ 
      NSInteger value1 = [[_dataStack pop] integerValue];
```
This amount of boilerplate made me think: Is there a way to generalize a binary function in C/ObjC without using `void*`?

File: Chime/Instructions.m [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    NSInteger value2 = [[_dataStack pop] integerValue];
+    [_dataStack push:@(value1 | value2)];
+    return;
+}
+
+void instruction_op_not(Stack *_dataStack){ NSInteger value = [[_dataStack pop] integerValue];
```
Comment (EduardoLR10) [inline Chime/Instructions.m]:
```suggestion
void instruction_op_not(Stack *_dataStack){ 
      NSInteger value = [[_dataStack pop] integerValue];
```

File: Chime/Headers/Instructions.h [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+void instruction_op_load_a(NSMapTable* _registers, Stack *_dataStack, NSMutableArray *_ram);
+void instruction_op_store_a(NSMapTable* _registers, Stack *_dataStack, NSMutableArray *_ram);
+void instruction_op_and(Stack *_dataStack);
+void instruction_op_or(Stack *_dataStack);
+void instruction_op_not(Stack *_dataStack);
+void instruction_op_xor(Stack *_dataStack);
+void instruction_op_plus(Stack *_dataStack);
+void instruction_op_double(Stack *_dataStack);
+void instruction_op_half(Stack *_dataStack);
+void instruction_op_plus_star(Stack *_dataStack);
```
Comment (EduardoLR10) [inline Chime/Headers/Instructions.h]: Is there any particular reason we should not make those methods of the VM?

**Review by EduardoLR10 — CHANGES_REQUESTED** [review]

File: Chime/Headers/Instructions.h (hunk 3 above)
Comment (z-silver) [reply Chime/Headers/Instructions.h]: We probably could, actually. I don't see any reason not to.

File: Chime/Instructions.m [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    [_dataStack push:@(!value)];
```
Comment (z-silver) [inline Chime/Instructions.m]:
```suggestion
    [_dataStack push:@(~value)];
```
This was my bad, `!` is actually boolean negation. The bitwise not is actually `~`.

**Review by z-silver — CHANGES_REQUESTED** [review]

File: Chime/Headers/Instructions.h (hunk 3 above)
Comment (MMagueta) [reply Chime/Headers/Instructions.h]: Already done in https://github.com/Dr-Nekoma/chime/pull/7

File: Chime/Instructions.m (hunk 4 above)
Comment (MMagueta) [reply Chime/Instructions.m:42]: Also solved in #7

**Review by z-silver — CHANGES_REQUESTED** [review]
Please add the directory `Chime.xcodeproj` to gitignore. Looks good otherwise.

**Review by EduardoLR10 — APPROVED** [review]

**Review by z-silver — APPROVED** [review]

**Pull request merged**